tekton/resources/prow-config-cd.yaml

```diff
@@ -0,0 +1,121 @@
+apiVersion: tekton.dev/v1alpha1
+kind: TriggerBinding
+metadata:
+  name: trigger-to-deploy-prow-config
+spec:
+  params:
+  - name: gitRepository
+    value: $(body.gitRepository)
+  - name: gitRevision
+    value: $(body.gitRevision)
+  - name: configPath
+    value: $(body.configPath)
+  - name: namespace
+    value: $(body.namespace)
+  - name: clusterResource
+    value: $(body.clusterResource)
+---
+apiVersion: tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+  name: prow-config-deployer
+spec:
+  serviceAccountName: release-right-meow
+  triggers:
+    - name: trigger
+      binding:
+        name: trigger-to-deploy-prow-config
+      template:
+        name: deploy-prow-config
+---
+apiVersion: tekton.dev/v1alpha1
+kind: TriggerTemplate
+metadata:
+  name: deploy-prow-config
+spec:
+  params:
+  - name: gitRepository
+    description: URL of the repository that holds the prow configuration
+  - name: gitRevision
+    description: Git revision
+  - name: namespace
+    description: Namespace to deploy to in the target cluster
+  - name: configPath
+    description: Path in the git repo that holds prow configs
+  - name: clusterResource
+    description: Name of the cluster resource that points to the target cluster
+  resourcetemplates:
+  - apiVersion: tekton.dev/v1alpha1
+    kind: PipelineResource
+    metadata:
+      name: git-source-$(uid)
+    spec:
+      type: git
+      params:
+      - name: revision
+        value: $(params.gitRevision)
+      - name: url
+        value: https://$(params.gitRepository)
+  - apiVersion: tekton.dev/v1alpha1
+    kind: TaskRun
+    metadata:
+      generateName: deploy-prow-config-$(uid)-
+    spec:
+      taskSpec:
+        inputs:
+          params:
+          - name: configPath
+            description: Path in the git repo that holds prow configs
+          - name: namespace
+            description: Namespace to deploy to in the target cluster
+          resources:
+          - name: source
+            type: git
+          - name: prow-cluster
+            type: cluster
+        stepTemplate:
+          env:
+          - name: KUBECONFIG
+            value: /workspace/$(inputs.resources.prow-cluster.name)/kubeconfig
+        steps:
+        - name: fetch-current-config
+          image: gcr.io/tekton-releases/dogfooding/kubectl
+          command:
+          - /bin/sh
+          args:
+          - -ce
+          - |
+            kubectl get configmap config -o template --template='{{ index .data "config.yaml" }}' \
```

Comment:
i wonder if its worth adding `set -e` and `set -x` in all the places where we're using `sh`? (just in case)

(or maybe you're waiting to use script mode!)

Reply:
Yeah, I was debating whether to use script mode already, but we probably should, and keep using the latest tech from Tekton :)
`-e` is already there, but `-x` would be a great idea. Is that automatically set in script mode?

Reply:
Done